share/functions/__fish_git_prompt.fish

```diff
@@ -417,6 +417,14 @@ function __fish_git_prompt --description "Prompt function for Git"
         set u "$___fish_git_prompt_color_untrackedfiles$u$___fish_git_prompt_color_untrackedfiles_done"
     end
     set b (string replace refs/heads/ '' -- $b)
+     if test -z "$__fish_git_prompt_branch_max_length"
+        set __fish_git_prompt_branch_max_length 20
+    end
+    set -l b_ori_length (string length $b)
```

Comment:
These probably don't need `--` since a branch name starting with `-` is invalid (`git branch -- -something` says so).

It wouldn't nurt, but it's probably okay to leave it out.